### DISCUSSION TITLE
git auto commit failing due to Your local changes to the following files would be overwritten by checkout

### DISCUSSION THREAD

**Discussion by dayananda30** [stefanzweifel/git-auto-commit-action#218] [Help]
Is anyone facing git auto commit failing due to Your local changes to the following files would be overwritten by checkout error even after using --force push option? 

I am using latest version - v4.14.1

```
error: Your local changes to the following files would be overwritten by checkout:
manifest_files/azure/azure-India/manifest.yaml
Please commit your changes or stash them before you switch branches.
Aborting
Error: Invalid status code: 1
at ChildProcess. (/__w/_actions/stefanzweifel/git-auto-commit-action/v4/index.js:17:19)
at ChildProcess.emit (events.js:[31](https://github.com/greenqloud/production-deployment-files/runs/6349307047?check_suite_focus=true#step:7:31)4:20)
at maybeClose (internal/child_process.js:1022:16)
at Process.ChildProcess._handle.onexit (internal/child_process.js:287:5) {
code: 1
}
Error: Invalid status code: 1
at ChildProcess. (/__w/_actions/stefanzweifel/git-auto-commit-action/v4/index.js:17:19)
at ChildProcess.emit (events.js:314:20)
at maybeClose (internal/child_process.js:1022:16)
at Process.ChildProcess._handle.onexit (internal/child_process.js:287:5
```

**Comment by BrunoJuchli** [comment]
Since you also created an issue for this (which got resolved, turns out it was a config error) it would have been nice if you would have at least linked to it:

https://github.com/stefanzweifel/git-auto-commit-action/issues/217

**Reply by stefanzweifel** [reply]
@BrunoJuchli Anything that we could improve in the documentation so that others don't run into this issue?

**Reply by BrunoJuchli** [reply]
@stefanzweifel 
The reported case here (see #217) is a rather obvious config mistake. I'm pretty sure that when one reads the documentation and follows it this would not happen - or only when it's an honest mistake, but then the docu doesn't help either.

Our case the same message "Your local changes to the following files would be overwritten by checkout" is triggered by a different issue - it's only occurring on some of the changes the linter creates (sometimes it works, sometimes it doesn't). We're still using v4 and I think this was fixed in 5.0.

Therefore, the only thing I can image could help is a library of error messages and possible solutions for it.
This would only help if google would pick it up, of course, and would compete with other resources like stack overflow, git hub issues and discussions, etc. I'm not really convinced this would be worth the effort.

Now, that I think of, it would be a very nice world if this library of possible solutions would be integrated into megalinter, and it would automatically parse the git error and propose solutions for it.
I'm asking way too much, I know ;-) So thanks for asking but I don't think there's something with a reasonable effort to be done.

**Reply by stefanzweifel** [reply]
A linter that automatically solves the issue for you is probably out of reach.

But better "error" messages might be possible.
Like when you get "Your local changes to the following files would be overwritten by checkout", the Action could output possible solutions:

```
- Check your configuration
- Check out this solution here https://github.com/…
- Try running `git pull` before running this Action
```

Can't promise anything, but the seed has been planted.
